### DISCUSSION TITLE
how to open a file by url

### DISCUSSION THREAD

**Discussion by NamelessAshone** [coder/code-server#3992] [Q&A] [answered]
Can I open a file by url in broswer?

**Comment by lg-nineoneninethreenine** [comment]
Can you elaborate?

**Reply by NamelessAshone** [reply]
In Windows and macOS, I can open files using the OS' URL handling mechanism. Use the following URL format:
```
vscode-insiders://file/FULL/PATH/TO/FILE
```

Is there a similar mechanism in code-server that can open a known file through url?

**Reply by lg-nineoneninethreenine** [reply]
well, idk about that but i will ping:

@jsjoeio 
@code-asher

**Reply by lg-nineoneninethreenine** [reply]
I don't think so, about that. But you can use the `Ctrl+O` mechanism to open files too.

**Reply by NamelessAshone** [reply]
If code-server supports access to files via url, users can embed vscode links into websites or blogs.

**Reply by lg-nineoneninethreenine** [reply]
> If code-server supports access to files via url, users can embed vscode links into websites or blogs.

That ain't possible, chief. Because, it redirects them to code-server, not VSCode. Moreover, it is problematic.

**Reply by lg-nineoneninethreenine** [reply]
If you tag a file which is NOT on their device or is on a different directory, it is hard. Moreover, code-server's main point is for coding from anywhere, but doesn't allow sharing. If you want a shared code-server, you need a server. Or you can use Gitpod for urgent envs.

**Comment by lg-nineoneninethreenine** [comment]
> Can I open a file by url in broswer?  

But, if you enter it as a query, it might be successful.

**Comment by jsjoeio** [comment, marked as answer]
No, this is currently not supported. See https://github.com/cdr/code-server/issues/1964

> We have ?folder= and ?workspace= but nothing for individual files yet.

https://github.com/cdr/code-server/issues/1964#issuecomment-741912748
